R/layers.R

```diff
@@ -51,27 +51,39 @@
 #' )
 #' }
 emodnet_get_layers <- function(wfs = NULL, service = NULL, service_version = "2.0.0",
-                               layers, crs = NULL, cql_filter = NULL,
-                               reduce_layers = FALSE, suppress_warnings = FALSE) {
+    layers, crs = NULL, cql_filter = NULL,
+    reduce_layers = FALSE, suppress_warnings = FALSE) {
 
     # check wfs ----------------------------------------------------------------
 
     if (is.null(wfs) & is.null(service)) {
         usethis::ui_stop(
-        "Please provide a valid {usethis::ui_field('service')} name or {usethis::ui_field('wfs')} object.
+            "Please provide a valid {usethis::ui_field('service')} name or {usethis::ui_field('wfs')} object.
          Both cannot be {usethis::ui_value('NULL')} at the same time."
         )
     }
 
-    if (is.null(wfs)) {
-        wfs <- emodnet_init_wfs_client(service, service_version)
-    }
+    wfs <- wfs %||% emodnet_init_wfs_client(service, service_version)
 
     check_wfs(wfs)
 
     # check layers -----------------------------------------
-    layers <- match.arg(layers, several.ok = TRUE,
-                       choices = emodnet_get_wfs_info(wfs)$layer_name)
+    layers <- match.arg(
+        layers,
+        several.ok = TRUE,
+        choices = emodnet_get_wfs_info(wfs)$layer_name
+    )
+
+    formats <- purrr::map_chr(layers, get_layer_format, wfs)
+    if (any(formats != "sf") && reduce_layers) {
```

Comment:
Hey @maelle ! So currently when reducing layers isn't possible, a list of each layer is returned with a warning instead of aborting all together (so as not to lose data that has already been downloaded). Should we not do the same in this situation too?

PS don't push any changes to the branch as I'm in the process of handling merge conflicts but can make the edit on my side)

Reply:
Oh you are correct.

Now how about we still abort, but somehow cache the data (e.g. with memoise to a temporary directory) so re-downloading would be fine? :thinking:

Reply:
Actually, this check is performed before downloading anything so I think it's not a problem to abort. In terms of caching, ultimately that would mean them having to rerun but with `reduce_layers = FALSE` which although makes the output more explicit, not sure adds to usability but would require more effort. Not sure at this point but could be convinced otherwise.  Do you have a strong preference either way?

Reply:
I prefer to see the code aborting when something isn't right (if that was your question).

Especially if the error happens before download (good catch :sweat_smile: )